modules/tinymce/src/core/test/ts/browser/bookmark/BookmarksTest.ts

```diff
@@ -223,4 +223,61 @@ describe('browser.tinymce.core.bookmark.BookmarksTest', () => {
     assertApproxRawContent(editor, '<p>abc</p>');
     TinyAssertions.assertSelection(editor, [ 0, 0 ], 1, [ 0, 0 ], 2);
   }));
+
+  describe('TINY-7817', () => {
+    const hook = TinyHooks.bddSetupLight<Editor>({
+      base_url: '/project/tinymce/js/tinymce'
+    }, [], false);
+
+    const outsideButton: SugarElement<HTMLButtonElement> = SugarElement.fromHtml('<button id="getBookmarkButton">Get Bookmark</button>');
+
+    before(() => {
+      Insert.append(SugarBody.body(), outsideButton);
+    });
+
+    after(() => {
+      Remove.remove(outsideButton);
+    });
+
+    it('TINY-7817: bookmark should be insert correclty even if the selection is on a comment', () => {
+      const editor = hook.editor();
+
+      editor.resetContent('<div><!-- Whatever --> <img></div>');
+      editor.addCommand('getBookmarkProxyCommand', () => {
```

Comment:
since the difference between the two tests was only the `content` I extracted a function that does all other operations(except for the expectation), I'm not sure, if I should put: `Insert.append(SugarBody.body(), outsideButton);` and `Remove.remove(outsideButton);` in the function too or if it could be useful to have it also in other tests with a logic different from the one in the function